src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -37,15 +38,23 @@ final class ArgumentResolver implements ArgumentResolverInterface
     public function __construct(ArgumentMetadataFactoryInterface $argumentMetadataFactory = null, iterable $argumentValueResolvers = [])
     {
         $this->argumentMetadataFactory = $argumentMetadataFactory ?? new ArgumentMetadataFactory();
-        $this->argumentValueResolvers = $argumentValueResolvers ?: self::getDefaultArgumentValueResolvers();
+        $this->argumentValueResolvers = $argumentValueResolvers ?: iterator_to_array(self::getIndexedDefaultArgumentValueResolvers());
     }
 
     public function getArguments(Request $request, callable $controller, \ReflectionFunctionAbstract $reflector = null): array
     {
         $arguments = [];
 
         foreach ($this->argumentMetadataFactory->createArgumentMetadata($controller, $reflector) as $metadata) {
-            foreach ($this->argumentValueResolvers as $resolver) {
+            if ($attributes = $metadata->getAttributesOfType(ValueResolver::class)) {
+                $resolverId = reset($attributes)->id;
+            }
+
+            foreach ($this->argumentValueResolvers as $id => $resolver) {
```

Comment:
This still iterates over the first value resolvers to skip them.
Can't we accept a ServiceProviderInterface for argumentValueResolvers and use that instead?

Reply:
I thought about it but I wanted the simplest implementation possible first. Will give it a try.